examples/prelude.pi

```diff
@@ -2,12 +2,22 @@
 record {
     id = fun A a => a,
     always = fun A B a b => a,
-    compose = fun A B C a-b b-c a => b-c (a-b a),
+
     dep-compose = fun A B C a-b a-b-c a => a-b-c a (a-b a),
-    flip = fun A B C a-b-c a b => a-b-c b a,
+    compose = fun A B C a-b b-c a => b-c (a-b a),
+    -- FIXME: Variable binding bug
+    -- compose = fun A B C => dep-compose A (fun a => B) (fun a b => C),
+    -- compose = fun A B C => dep-compose A (always B) (always (always C)),
```

Comment:
This uncovered a variable binding bug that was added in 1afb4761667825afb5b380534392f4450194addd - serves me right for not having a more comprehensive test suite! 🤦‍♂️ 

It means that dependent functions are completely broken:
```
> (fun A a => a : Fun (A : Type) -> A -> A)
fun A a => a : Fun (A : Type) -> Fun (t : A) -> A
> (fun A a => a : Fun (A : Type) -> A -> A) S32
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', pikelet/src/lang/core/semantics.rs:481:52
```
I'll have to track this down in a later PR, but for now I've commented this stuff out.

Reply:
I've posted more details here: https://github.com/pikelet-lang/pikelet/pull/229#issuecomment-706497438